plugins/rbac-backend/src/helpers/csv.ts

```diff
@@ -0,0 +1,200 @@
+import { Enforcer, FileAdapter, newEnforcer, newModelFromString } from 'casbin';
+import { isEqual } from 'lodash';
+
+import { MODEL } from '../service/permission-model';
+import {
+  validateEntityReference,
+  validatePolicy,
+} from '../service/policies-validation';
+import {
+  checkForDuplicateGroupPolicies,
+  checkForDuplicatePolicies,
+  transformArraytoPolicy,
+  updatePolicies,
+} from './utils';
+
+export const loadFilteredPoliciesFromCSV = async (
+  policyFile: string,
+  enf: Enforcer,
+  entityRef: string,
+  resourceType: string,
+  action: string,
+) => {
+  const fileEnf = await newEnforcer(
+    newModelFromString(MODEL),
+    new FileAdapter(policyFile),
+  );
+
+  const roles = await enf.getFilteredGroupingPolicy(0, entityRef);
+
+  const policies: string[][] = [];
+  const enforcerPolicies: string[][] = [];
+
+  for (const role of roles) {
+    const policyByRole = await fileEnf.getFilteredPolicy(
+      0,
+      role[1],
+      resourceType,
+      action,
+    );
+    policies.push(...policyByRole);
+    const enforcerPolicy = await enf.getFilteredPolicy(
+      0,
+      role[1],
+      resourceType,
+      action,
+    );
+    enforcerPolicies.push(...enforcerPolicy);
+  }
+
+  if (isEqual(policies, enforcerPolicies)) {
+    return;
+  }
+
+  if (policies.length === 0) {
+    policies.push(
+      ...(await fileEnf.getFilteredPolicy(1, resourceType, action)),
+    );
+  }
+
+  for (const policy of policies) {
+    const err = validatePolicy(transformArraytoPolicy(policy));
+    if (err) {
+      throw new Error(
```

Comment:
I think we should throw warning at least for now, because broken policies file will make each permission validation request not working. Also we can implement warning events for frontend plugin in the feature.